apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryInsertResultsTests.cs

```diff
@@ -0,0 +1,426 @@
+                yield return new object[] { new DerivedBigQueryClient(), new InsertOptions(), new List<BigQueryInsertRow>(), null };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NotNullableConstructorParametersData))]
+        public void Constructor_NullParameters_Fail(
+            BigQueryClient client, InsertOptions options, IList<BigQueryInsertRow> rows, TableDataInsertAllResponse response) =>
+            Assert.Throws<ArgumentNullException>(() => new BigQueryInsertResults(client, options, rows, response));
+
+        [Fact]
+        public void Constructor_NullOptions() =>
+            Assert.NotNull(new BigQueryInsertResults(_client, null, new List<BigQueryInsertRow>(), new TableDataInsertAllResponse()));
+
+        public static IEnumerable<object[]> NoErrorsData
+        {
+            get
+            {
+                yield return new object[] { new TableDataInsertAllResponse() };
+                yield return new object[] { new TableDataInsertAllResponse { InsertErrors = new List<InsertErrorsData>() } };
+                yield return new object[] { new TableDataInsertAllResponse { InsertErrors = new List<InsertErrorsData> { null, null, null } } };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NoErrorsData))]
+        public void AllRowsInserted(TableDataInsertAllResponse response)
+        {
+            BigQueryInsertResults results = new BigQueryInsertResults(
+                _client, 
+                new InsertOptions(), 
+                new List<BigQueryInsertRow> { new BigQueryInsertRow() },
+                response);
+
+            Assert.Equal(BigQueryInsertStatus.AllRowsInserted, results.Status);
+            Assert.False(results.HasErrors);
+            Assert.Same(results, results.ThrowOnNoneInserted());
+            Assert.Same(results, results.ThrowOnNotAllInserted());
+            Assert.Same(results, results.ThrowOnAnyError());
+            Assert.Empty(results.Errors);
+        }
+
+        [Fact]
+        public void SomeRowsInserted()
+        {
+            ErrorProto row1Error1 = new ErrorProto { Location = "field_1", Reason = "reason_1", Message = "message_1" };
+            ErrorProto row1Error2 = new ErrorProto { Location = "field_2", Message = "message_2" };
+            InsertErrorsData row1List1 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error1, row1Error2 } };
+
+            ErrorProto row5Error1 = new ErrorProto { Location = "field_3", Reason = "reason_3" };
+            InsertErrorsData row5List1 = new InsertErrorsData { Index = 5, Errors = new List<ErrorProto> { row5Error1 } };
+
+            InsertErrorsData row6List1 = new InsertErrorsData { Index = 6 };
+
+            ErrorProto row1Error3 = new ErrorProto { Location = "field_4", Reason = "reason_4", Message = "message_4" };
+            InsertErrorsData row1List2 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error3 } };
```

Comment:
I don't currently understand how there would be two separate sets of errors from the same row. I hope I'll understand it later...